### ISSUE TITLE
Port the extension to GNOME 45

### ISSUE THREAD

**Issue by jeffshee** [jeffshee/gnome-ext-hanabi#81]
**Description**
This is the meta thread for the progress of porting the extension to GNOME 45.

**Progress**
- [x] [Port the extension to ESM](https://www.youtube.com/watch?v=Bcta80iNuQA)

**Additional Information**

**Comment by M4TH1EU** [comment]
Just tried the gnome-45 branch on Fedora 39 with Gnome 45. Seems to work alright so far, except there is a black bar on the right side of my screen (2560x1440, 100% scaling). Just though I would report it here.
Also a way to set a video per screen would be awesome as my second monitor is in vertical so the video uses 1/3 of the screen the rest being all black.
![Capture d’écran du 2023-10-24 20-08-28](https://github.com/jeffshee/gnome-ext-hanabi/assets/41210834/64c3b837-cac9-4dde-81fc-12a0cee844c7)

Great work so far ! Thanks.

**Comment by SyntheticBird45** [comment]
Hi, I would have like to say the same. I've been using the gnome-45 branch for 1 months now and it was fine.
There was just a bug where if you were in a window (or overview) just after gnome launched, the wallpaper would hide the top bar.
I updated my Arch yesterday and now it looks like this : 
![gnome45brancharchupdatedyesterday cleaned](https://github.com/jeffshee/gnome-ext-hanabi/assets/118022351/3361f053-dd42-40a6-996c-12087348580b)

I will have to stick with the shit Hidamari is, for the time being. I still love this extension tho.

**Comment by minpeter** [comment]
It loads normally in Arch and gnome45, but the video doesn't play.
*AUR was used and the source of the PKGBUILD file was arbitrarily manipulated and installed.

Hope to see hanabi on gnome 45 soon :)

**Comment by jeffshee** [comment]
@SyntheticBird45 I can confirm the issue in Fedora 39. This appears only in Wayland, seems like a weird bug in the shell. I'm trying to fix it now.

**Comment by jeffshee** [comment]
Hi @SyntheticBird45, I'm glad to hear that.

It turned out that it wasn't Hanabi extension's fault but rather a Mutter's bug in GNOME 45, which caused the "detached" wallpaper issue. I have reported it to Mutter, and hopefully, it will get fixed someday. (I managed to fix it with a workaround, but it might not be the ideal solution.)
https://gitlab.gnome.org/GNOME/mutter/-/issues/3159

Regarding the new bugs you mentioned, I couldn't reproduce the issue on my system, so I have no idea about that yet. :/
